test/ng/browserSpecs.js

```diff
@@ -567,6 +568,30 @@ describe('browser', function() {
           expect(replaceState).not.toHaveBeenCalled();
           expect(locationReplace).not.toHaveBeenCalled();
         });
+
+        it('should not do pushState with a URL only different in encoding', function() {
+          //A URL from something such as window.location.href
+          browser.url('http://server/abc?q=%27');
+
+          pushState.calls.reset();
+          replaceState.calls.reset();
+          locationReplace.calls.reset();
+
+          //A prettier URL from something such as $location
+          browser.url('http://server/abc?q=\'');
+          expect(pushState).not.toHaveBeenCalled();
+          expect(replaceState).not.toHaveBeenCalled();
+          expect(locationReplace).not.toHaveBeenCalled();
+        });
+
+        it('should not do pushState on $$checkUrlChange() with a URL only different in encoding', function() {
+          var callback = jasmine.createSpy('onUrlChange');
+          browser.onUrlChange(callback);
+          browser.url('http://server/abc?q=\'');
+
+          browser.$$checkUrlChange();
```

Comment:
Since we are testing a private API here, I feel like maybe we should test this in $location spec instead?

Reply:
To avoid invoking the private API from tests? It really is testing logic only in browser.js though :/